### PR TITLE
Homework10: Valeria Kudelko

### PR DESCRIPTION
**Pull request by ValeriaKudelko**

### CODE REVIEW

File: homework10part3.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+
+def typed(type):
+    """
+    Called function.
+    """
+    def decorator(func):
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            args = [convert_val(arg, type) for arg in args]
```
Comment (Stml89) [inline homework10part3.py]: `map()` is also could help here

File: homework10part4.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    Called function.
+    """
+    cache_save = {}
+
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        key = tuple(args) + tuple(kwargs.items())
+        if key not in cache_save:
+            cache_save[key] = func(*args, **kwargs)
+        return cache_save[key]
```
Comment (Stml89) [inline homework10part4.py]: A bit simpler solution:
```suggestion
    def wrapper(arg):
        if arg in cache_dictionary:
            return cache_dictionary[arg]
        cache_dictionary[arg] = func(arg)
        return cache_dictionary[arg]
```

**Review by Stml89 — APPROVED** [review]

File: homework10part4.py (hunk 2 above)
Comment (ValeriaKudelko) [reply homework10part4.py]: I agree, I changed it

File: homework10part3.py (hunk 1 above)
Comment (ValeriaKudelko) [reply homework10part3.py]: added, thank you

**Pull request merged**